drivers/ethernet/eth_stm32_hal_priv.h

```diff
@@ -1,34 +1,93 @@
 /*
  * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+ * Copyright (c) 2025 STMicroelectronics
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #ifndef ZEPHYR_DRIVERS_ETHERNET_ETH_STM32_HAL_PRIV_H_
 #define ZEPHYR_DRIVERS_ETHERNET_ETH_STM32_HAL_PRIV_H_
 
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
 #include <zephyr/kernel.h>
 #include <zephyr/types.h>
+#include <soc.h>
 
-/* Naming of the  ETH PTP Config Status changes depending on the stm32 serie */
-#if defined(CONFIG_SOC_SERIES_STM32F4X)
-#define ETH_STM32_PTP_CONFIGURED HAL_ETH_PTP_CONFIGURATED
-#define ETH_STM32_PTP_NOT_CONFIGURED HAL_ETH_PTP_NOT_CONFIGURATED
+#define DT_DRV_COMPAT st_stm32_ethernet
+
+#define LOG_LEVEL CONFIG_ETHERNET_LOG_LEVEL
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+extern struct eth_stm32_hal_dev_data eth0_data;
```

Comment:
I don't like this being shared over multiple files, should be set near `ETH_NET_DEVICE_DT_INST_DEFINE` in preparation for adding support for multiple instance support

Reply:
Done